sqlmodel/main.py

```diff
@@ -414,7 +415,7 @@ def get_column_from_field(field: ModelField) -> Column:
         return sa_column
     sa_type = get_sqlachemy_type(field)
     primary_key = getattr(field.field_info, "primary_key", False)
-    nullable = not field.required
+    nullable = not primary_key and _is_field_nullable(field)
```

Comment:
Should this happen after the `if hasattr(field.field_info, "nullable"):` block below?
Does it make sense to allow people to still set `nullable=True` if it was set to `False` here because of the field being a primary key?

Reply:
Good point! Thanks! :bow: :rocket:

Reply:
Sorry, completely missed your message. You are right, thanks for the help!

Reply:
Hehe it's still not super clear and obvious how this should work, I moved it and then that caused https://github.com/tiangolo/sqlmodel/issues/420 :sweat_smile:

Anyway, I'm on it. :nerd_face: